pandas/core/frame.py

```diff
@@ -3338,12 +3338,10 @@ def transpose(self, *args, copy: bool = False) -> DataFrame:
         ):
             # We have EAs with the same dtype. We can preserve that dtype in transpose.
             dtype = dtypes[0]
-            arr_type = dtype.construct_array_type()
             values = self.values
 
-            new_values = [arr_type._from_sequence(row, dtype=dtype) for row in values]
             result = self._constructor(
-                dict(zip(self.index, new_values)), index=self.columns
+                values.T, index=self.columns, columns=self.index, dtype=dtype
```

Comment:
values.T is going to cast to ndarray, which often means object

i think could use DataFrame.from_arrays?

Reply:
I see a MultiIndex.from_arrays, but not DataFrame. Here are some timings:

```
size = 1000
df = pd.DataFrame({'a': size * [1, 2, 3]}, dtype='category')
df_T = df.T

%timeit df.T
%timeit df_T.T
```

PR:

```
249 ms ± 6.39 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
92.3 ms ± 6.51 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

Master:

```
237 ms ± 4.78 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
89.7 ms ± 6.87 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

so certainly a slight regression. Will continue looking to see if avoiding object is possible.

Reply:
Ah, found it - `_from_arrays`.

Reply:
New timing with the update

```
234 ms ± 4.46 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
85 ms ± 4.8 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```